hier_config/platforms/view_base.py

```diff
@@ -0,0 +1,271 @@
+        if self.tagged_all:
+            return InterfaceDot1qMode.TAGGED_ALL
+        if self.tagged_vlans:
+            return InterfaceDot1qMode.TAGGED
+        if self.native_vlan and not self.is_svi:
+            return InterfaceDot1qMode.ACCESS
+        return None
+
+    @property
+    @abstractmethod
+    def duplex(self) -> InterfaceDuplex:
+        """Determine the configured Duplex of the interface."""
+
+    @property
+    @abstractmethod
+    def enabled(self) -> bool:
+        """Determines if the interface is enabled."""
+
+    @property
+    @abstractmethod
+    def has_nac(self) -> bool:
+        """Determine if the interface has NAC configured."""
+
+    @property
+    def ipv4_interface(self) -> IPv4Interface | None:
+        """Determine the first configured IPv4Interface, address/prefix, object."""
+        return next(iter(self.ipv4_interfaces), None)
+
+    @property
+    @abstractmethod
+    def ipv4_interfaces(self) -> Iterable[IPv4Interface]:
+        """Determine the configured IPv4Interface, address/prefix, objects."""
+
+    @property
+    @abstractmethod
+    def is_bundle(self) -> bool:
+        """Determine if the interface is bundle."""
+
+    @property
+    @abstractmethod
+    def is_loopback(self) -> bool:
+        """Determine if the interface is loopback."""
+
+    @property
+    def is_subinterface(self) -> bool:
+        """Determine if the interface is a subinterface."""
+        return "." in self.name
+
+    @property
+    @abstractmethod
+    def is_svi(self) -> bool:
+        """Determine if the interface is an SVI."""
+
+    @property
+    @abstractmethod
+    def module_number(self) -> int | None:
+        """Determine the module number of the interface."""
+
+    @property
+    @abstractmethod
+    def nac_control_direction_in(self) -> bool:
+        """Determine if the interface has NAC control direction in configured."""
+
+    @property
+    @abstractmethod
+    def nac_host_mode(self) -> NACHostMode | None:
+        """Determine the NAC host mode."""
+
+    @property
+    @abstractmethod
+    def nac_mab_first(self) -> bool:
+        """Determine if the interface has NAC configured for MAB first."""
+
+    @property
+    @abstractmethod
+    def nac_max_dot1x_clients(self) -> int:
+        """Determine the max dot1x clients."""
+
+    @property
+    @abstractmethod
+    def nac_max_mab_clients(self) -> int:
+        """Determine the max mab clients."""
+
+    @property
+    @abstractmethod
+    def name(self) -> str:
+        """Determine the name of the interface."""
+
+    @property
+    @abstractmethod
+    def native_vlan(self) -> int | None:
+        """Determine the native VLAN."""
+
+    @property
+    @abstractmethod
+    def number(self) -> str:
+        """Remove letters from the interface name, leaving just numbers and symbols."""
+
+    @property
+    @abstractmethod
+    def parent_name(self) -> str | None:
+        """Determine the parent bundle interface name."""
```

Comment:
NitPick: Some of these docstrings have periods, but may not be sentences... Consider removing the periods or making them complete sentences.

Reply:
I gave it a once over and made improvements.